### ISSUE TITLE
messaging – io.js is much more than just es6.

### ISSUE THREAD

**Issue by timoxley** [nodejs/evangelism#38]
**tl;dr;** rather than riding on the laurels of es6 support in v8, I'd like to see io.js find a way to tactfully communicate that io.js is a good, safe bet for business and definitely not "hot, new unstable tech".

This is probably obvious but in my quick skimming of the issue queue I didn't see any discussion about it.

![image](https://cloud.githubusercontent.com/assets/43438/6540991/4c7d8862-c4f5-11e4-928f-73174fae32ee.png)

"Bringing es6 to node" isn't doing io.js enough justice. io.js is doing so much more than just getting a few new JavaScript features, in-fact the efforts going into io.js is everything _except_ the es6 features – that was just a nice side-effect of updating v8, and now it's being used as the primary messaging?
#### "unstable" & "experimental"

A significant perception problem io.js has at the moment is that it's seen as the "unstable", "experimental" version of node. es6 itself is still seen as "unstable" and "experimental" since the spec has been in flux and support in various engines a bit rocky. Focusing on the es6 support rather than all the other awesome shit _might_ be compounding the perception problem.
#### "If it ain't broke, don't fix it."

Rapid iteration is important so bugs are squashed quickly and additional features be laid, but the the more conservative is more likely to focus on the corollary: more opportunities to introduce bugs and more APIs changing requiring rewrites to working code. Would be good to flip this around and make it clear that this should mean that there are less bugs, not more.

One of the first benefits i see communicated is "chrome doesn't even support that version of v8". This doesn't mean anything to most developers as it's hard to imagine what the downsides are of an unsupported v8. An embargoed security update would be a gold medal for concrete examples to demonstrate this. GOLD MEDAL.

Also might be good to clear up how this fits in the idea that one day node would be "done".
#### trickle down effects

I realise that the messaging is politically sensitive, and perhaps this is the best of a bad bunch. It's clear though that es6 was/is probably the quickest way to get developers excited about io.js – people don't generally get as piqued up about escaping a corporate dictatorship and implementing new governance models as they do about "exciting new ways to define a string". es6 may not a continue to be a good differentiator though – given that people see io.js as the new unstable branch of node, as it's conceivable that many are over-confident that the es6 features will eventually trickle downstream into node.

I'd like to suggest focusing on more "objective", quantitative metrics like user engagement, number of bugs fixed, number of downloads, benchmarks, state of the CI suite, etc. 

Sadly I don't have any concrete suggestions for a new tagline.

**Comment by mikeal** [comment]
This was always intended to be the "initial messaging" and we would change this over time, possibly in conjunction with the next major release.

We had figured that it would take longer than it did to stabilize and we also didn't anticipate that 0.12.0 would release and be much more unstable by comparison, so this messaging got out of touch with the project much quicker than anticipated.

One concern I have is that we don't have any _huge_ points of differentiation, other than ES6. Messaging "stability" in a way that differentiates us is difficult and potentially political ridden (see: jsdom's remarks about node.js being "legacy").

**Comment by blakmatrix** [comment]
"People don't buy what you do, they buy why you do it"(Simon Sinek). I firmly believe this statement to be true. 

Node and the community around it were brought to life around the notion of non-blocking IO, and since I've been following node forward and io.js I've believed it to be simply an extension of that original idea, non-blocking governance. 

I know metrics are great and can decidely convert business, but at the core and always asked is why? Why use node, why use io.js?

As @mikeal said, other than ES6 of the question, "what's different?" There isn't anything major to note, but I disagree. The message of "Why is it different?" Is the major difference, and the ultimate messaging. If we were to distill 'the why' into the tagline, it should have the desired effect. 

As a warning or a flag, while it might be necessary in the short run, focusing on comparisons will do us all a disservice in the end. The focus needs to be where we are going vs where we've been, what's left to go vs how much we've already achieved. "Bringing Node Forward" VS "the future of [non blocking] I/O". If we're going to continue to compare io.js to node, we are most likely going to be stuck being a fancy fork of node.

**Comment by bnb** [comment]
> As a warning or a flag, while it might be necessary in the short run, focusing on comparisons will do us all a disservice in the end. The focus needs to be where we are going vs where we've been, what's left to go vs how much we've already achieved.

+1. I think focusing on the future of the project is a way to separate ourselves from the issues @mikeal and @timoxley raised about messaging. People have Node as a frame of reference for what io.js is doing, so we need to find a way to move that up and have io.js be its own reference. We've pretty much moved on from the major Node issues - governance, updates, bugfixes, community involvement, etc. - and I think now might be a good time to start changing our focus to what io.js brings the developer community as a whole, not what io.js does to improve on Node.

**Comment by mikeal** [comment]
"JavaScript platform for... well... everything!"

**Comment by timoxley** [comment]
> One concern I have is that we don't have any huge points of differentiation, other than ES6.

https://github.com/iojs/io.js/blob/v1.x/CHANGELOG.md

That's pretty huge.

Could also just push the roadmap harder… allows you to cash in on features you haven't even implemented yet. May feel disingenuous but a real part of what io.js is selling is the vision.

**Comment by timoxley** [comment]
"Get involved now and help shape the future of serverside JavaScript."
vs
"don't get your hopes up"

**Comment by yosuke-furukawa** [comment]
More simple and strong word.

"Open up the future of JavaScript platform"

**Comment by timoxley** [comment]
> "Open up the future of JavaScript platform"

I like the direction.

**Comment by mikeal** [comment]
+1 on this "future of" direction.

**Comment by blakmatrix** [comment]
Some Ideas:
- The future of I/O
- The future of reliability
- The future of communication
- The future of stability
- The future of collaboration 
- `this` is the future of I/O
- `this` is the future of open scalability

**Comment by rstacruz** [comment]
Why not just shorten that to something as simple as this?

> "The future of JavaScript"

**Comment by yosuke-furukawa** [comment]
I like the simple message.
However, "The future of JavaScript" misleads that we will create next JavaScript specs like TC39.

According to nodejs.org site, they tell me what node.js is. 

> Node.js is a platform 

So I suggest "The future of JavaScript Platform".

**Comment by julianduque** [comment]
The future of server-side JavaScript

**Comment by vjk2005** [comment]
**A JavaScript platform for the people, by the people.**

Then below the tagline is the list of all the features like ES6, Open-ness etc.

**Comment by yosuke-furukawa** [comment]
IMO, io.js and node.js are not only server-side one.
We have NW.js and AtomShell. We use gulp / grunt task runner in our PC.
They are JS executable environments and JS platforms for us.

**Comment by rstacruz** [comment]
:+1: on moving away from the "server-side" wording.

**Comment by julianduque** [comment]
The JavaScript platform of the Future

**Comment by a0viedo** [comment]
> The JavaScript platform of the Future

+1

**Comment by megastef** [comment]
.next - might get likes from MS users ;)

**Comment by bnb** [comment]
Was a final conclusion made with this?

**Comment by mikeal** [comment]
We'll need to revisit after the foundation merge, this can get closed.
